### ISSUE TITLE
Qwen-moe as actor model, and using more than one gpu, will hang during training 

### ISSUE THREAD

**Issue by JiruWang19971997** [OpenRLHF/OpenRLHF#1063]
environment: 
Openrlhf: latest main branch
vllm: latest
cuda: 12.4
deepspeed: 0.16.7
training script```
set -x 
ray job submit --address="http://127.0.0.1:8267" \
   --runtime-env-json='{"working_dir": "/root/OpenRLHF"}' \
   -- python3 -m openrlhf.cli.train_ppo_ray \
    --ref_num_nodes 1 \
    --ref_num_gpus_per_node 1 \
    --reward_num_nodes 1 \
    --reward_num_gpus_per_node 4 \
    --critic_num_nodes 1 \
    --critic_num_gpus_per_node 1 \
    --actor_num_nodes 1 \
    --actor_num_gpus_per_node 2 \
    --vllm_num_engines 1 \
    --vllm_tensor_parallel_size 1 \
    --pretrain qwen1.5-moe \
    --reward_pretrain qwen-rm \
    --micro_train_batch_size 1 \
    --train_batch_size 32 \
    --micro_rollout_batch_size 2 \
    --rollout_batch_size 1024 \
    --max_epochs 1 \
    --prompt_max_len 1024 \
    --generate_max_len 1024 \
    --zero_stage 2 \
    --bf16 \
    --actor_learning_rate 5e-7 \
    --critic_learning_rate 9e-6 \
    --init_kl_coef 0.01 \
    --prompt_data data \
    --input_key question \
    --apply_chat_template \
    --max_samples 10000000 \
    --packing_samples \
    --normalize_reward \
    --adam_offload \
    --flash_attn \
    --vllm_sync_backend gloo \
    --gradient_checkpointing 

When training both the actor and critic models, the critic model performs well. However, the actor model with Moe architecture is very slow during backward propagation. Especially, when step 1 is finished, step 2 will never be launched, causing a timeout error.

**Comment by hijkzzz** [comment]
this is a bug of deepspeed https://github.com/deepspeedai/DeepSpeed/issues/7461

**Comment by hijkzzz** [comment]
https://github.com/OpenRLHF/OpenRLHF/commit/d5fcb42bff8e2ea8607367289e9b82288cd065c0

**Comment by Aurelius84** [comment]
I got same problem, Qwe3 MoE + Zero2 hang， but Qwen MoE + Zero3 was not hang....
